static/app/utils/event/useEventCanShowReplayUpsell.tsx

```diff
@@ -56,10 +56,14 @@ export default function useEventCanShowReplayUpsell({
     ? getConfigForIssueType(group, group?.project).pages.replays.enabled
     : true;
 
+  // Only show the upsell if the issue & project supports replays,
+  // the org has never sent replays, and
+  // the project platform is not a backend platform.
   const canShowUpsell =
     groupHasReplays &&
     projectCanUpsellReplay(project) &&
-    (!hasOrgSentReplays || replayBackendPlatforms.includes(upsellPlatform));
+    !hasOrgSentReplays &&
+    !replayBackendPlatforms.includes(upsellPlatform);
```

Comment:
Can we integrate this `replayBackendPlatforms` into `projectCanUpsellReplay` directly? it looks like `projectCanUpsellReplay()` is only called here, and it's doing `replayPlatforms.includes()` which looks at the list of frontend/backend/mobile platforms.